pkg/validations/base_test.go

```diff
@@ -85,13 +87,34 @@ func createTestDeployment(replicas int32) (*appsv1.Deployment, error) {
 	return &d, nil
 }
 
+func intializeEngine(t *testing.T, customCheck ...config.Check) {
+
+	// Reset global prometheus registry to avoid testing conflicts
+	metrics.Registry = prometheus.NewRegistry()
+
+	// Check if custom check has been set
+	if len(customCheck) > 0 {
+		// Initialize engine with custom check
+		e, err := newEngine(newEngineConfigWithCustomCheck(customCheck))
+		if err != nil {
+			t.Errorf("Error creating validation engine with custom checks %v", err)
```

Comment:
Better to return an error here and let the caller deal with it.  That would also remove the need to pass in the testing.T.